magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,253 @@
+
+                with open(f"{outputdir}/logs/list_LST.txt", "a+") as LSTdataPathFile:
+                    for subrun in list_of_subruns:
+                        LSTdataPathFile.write(f"{subrun}\n")
+
+                if not os.path.exists(f"{outputdir}/logs/list_LST.txt"):
+                    continue
+                with open(f"{outputdir}/logs/list_LST.txt", "r") as f:
+                    process_size = len(f.readlines()) - 1
+
+                if process_size < 0:
+                    continue
+                slurm = slurm_lines(
+                    queue="short",
+                    job_name=f"{source_name}_coincidence",
+                    array=process_size,
+                    mem="8g",
+                    out_name=f"{outputdir}/logs/slurm-%x.%A_%a",
+                )
+                rc = rc_lines(
+                    store="$SAMPLE ${SLURM_ARRAY_JOB_ID} ${SLURM_ARRAY_TASK_ID}",
+                    out="$OUTPUTDIR/logs/list",
+                )
+
+                lines = (
+                    slurm
+                    + [
+                        f"export INM={MAGIC_DL1_dir}/Merged/{d}\n",
+                        f"export OUTPUTDIR={outputdir}\n",
+                        "SAMPLE_LIST=($(<$OUTPUTDIR/logs/list_LST.txt))\n",
+                        "SAMPLE=${SAMPLE_LIST[${SLURM_ARRAY_TASK_ID}]}\n",
+                        "export LOG=$OUTPUTDIR/logs/coincidence_${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}.log\n",
+                        f"conda run -n {env_name} lst1_magic_event_coincidence --input-file-lst $SAMPLE --input-dir-magic $INM --output-dir $OUTPUTDIR --config-file {target_dir}/v{__version__}/{source_name}/config_coincidence.yaml >$LOG 2>&1\n",
+                    ]
+                    + rc
+                )
+                with open(
+                    f"{source_name}_LST_coincident_{outputdir.split('/')[-1]}.sh",
+                    "w",
+                ) as f:
+                    f.writelines(lines)
+
+
+def main():
+
+    """
+    Main function
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--config-file",
+        "-c",
+        dest="config_file",
+        type=str,
+        default="./config_general.yaml",
+        help="Path to a configuration file",
+    )
+
+    args = parser.parse_args()
+    with open(
+        args.config_file, "rb"
+    ) as f:  # "rb" mode opens the file in binary format for reading
+        config = yaml.safe_load(f)
+    target_dir = Path(config["directories"]["workspace_dir"])
+
+    env_name = config["general"]["env_name"]
+    LST_version = config["general"]["LST_version"]
+
+    source_in = config["data_selection"]["source_name_database"]
+    source = config["data_selection"]["source_name_output"]
+    cluster = config["general"]["cluster"]
+
+    source_list = []
+    if source_in is None:
+        source_list = joblib.load("list_sources.dat")
+
+    else:
+        source_list.append(source)
```

Comment:
shouldn't it be ```source_list.append(source_in)``` ?
however then ```source``` would not be used at all

the two source names are a bit confusing

Reply:
source_in is the name in the database, source is the 'tag' used for the source (and the name of the folder in which you store the results): e.g., source_in=CrabNebula, source=Crab_23

Reply:
but the check is if source is not None
so it might happen that somebody puts source_in=CrabNebula, but does not put source, and then you add None to the list. 
You could add a condition that if source is None you put source = source_in

Reply:
I will add it now in a small PR (the same 'structure' is used in many scripts)